### PR TITLE
Align Cicero Server API with Cicero CLI - Part 1

### PR DESCRIPTION
**Pull request by martinhalford**
# Issue #508 
Implemented /parse & /draft endpoints

### Changes
- implemented /parse endpoint
- implemented /draft endpoint
- moved repeated code to function initClauseTemplate(req) to improve code reuse.

### Flags
- The /invoke and other commands remain unimplemented.

### CODE REVIEW

**Comment by martinhalford** [comment]
Closing this pull request and submitting a new one in its place.

**Comment by martinhalford** [comment]
Requested changes made.
1. Refactored main app.js to remove global vars and have InitClauseInstance return a clause object instead.
2. Fixed ESLint issues.
3. Added (admittedly) a couple of very simple tests to test/app.js to check the new /parse and /draft endpoints.

File: packages/cicero-server/test/app.js [hunk 1]
```diff
@@ -88,6 +88,18 @@ describe('cicero-server', () => {
             });
     });
 
+    it('/should parse a simple stateless request (ergo)', async () => {
+        return request.post('/parse/latedeliveryandpenalty/data.json')
+            .send(body)
+            .expect(200);
```
Comment (jeromesimeon) [inline packages/cicero-server/test/app.js]: Could we see the expected response here (as in the `trigger` case) ?

File: packages/cicero-server/test/app.js [hunk 2]
```diff
@@ -88,6 +88,18 @@ describe('cicero-server', () => {
...
+    });
+
+    it('/should draft a simple stateless request (ergo)', async () => {
+        return request.post('/draft/latedeliveryandpenalty/data.json')
+            .send(body)
+            .expect(200);
```
Comment (jeromesimeon) [inline packages/cicero-server/test/app.js]: Could we see the expected response here (as in the `trigger` case) ?

**Comment by martinhalford** [comment]
Doh! Yes - I missed DCO. I have rectified and recommitted changes.  Thanks!

**Pull request merged**